SourceGenerators/CodeDependency.cs

```diff
@@ -0,0 +1,11 @@
+namespace GodotSharp.SourceGenerators;
+
+// We rely on the record equality here
+public abstract record CodeDependency
+{
+    public record SceneTree(string TscnFileName, bool TraverseInstancedScenes) : CodeDependency
+    {
+        public string ClassName { get; } = $"_{Path.GetFileNameWithoutExtension(TscnFileName).ToSafeName()}";
```

Comment:
Just realized, that this name should also depend on the `TraverseInstancedScenes` value, otherwise there will be a name conflict if there are two consumers who provide a different value for the same scene.